shared/lib_shared_inverter.cpp

```diff
@@ -45,232 +45,275 @@ SharedInverter::SharedInverter(const SharedInverter& orig){
+    std::vector<double> powerDC_Watts_one_inv;
+    std::vector<double> powerDC_Watts_one_inv_iter;
+    for (size_t i = 0; i < powerDC_kW_in.size(); i++) {
+        powerDC_Watts_one_inv.push_back(powerDC_kW_in[i] * util::kilowatt_to_watt / m_numInverters);
+        //powerDC_Watts_one_inv_iter[i] = powerDC_kW_in[i] * util::kilowatt_to_watt / m_numInverters;
+    }
+    Tdry_C = tempC;
     StringV = DCStringVoltage[0];
-    double tempLoss = 0.0;
-	if (m_tempEnabled){
-		//use average of the DC voltages to pick which temp curve to use- a weighted average might be better but we don't have that information here
-		double avgDCVoltage = 0;
-		for (size_t i = 0; i < DCStringVoltage.size(); i++)
-			avgDCVoltage += DCStringVoltage[i];
-		avgDCVoltage /= DCStringVoltage.size();
-		calculateTempDerate(avgDCVoltage, tempC, powerAC_Watts, efficiencyAC, tempLoss);
-	}
-
-	// Scale to total system size
-	// Do not need to scale back up by m_numInverters because scaling them down was a separate vector, powerDC_Watts_one_inv
-	powerDC_kW = 0;
-	for (size_t i = 0; i < powerDC_kW_in.size(); i++)
-		powerDC_kW += powerDC_kW_in[i];
-
-	//Convert units to kW and scale to total array for all other outputs
-	convertOutputsToKWandScale(tempLoss, powerAC_Watts);
+    int size = DCStringVoltage.size();
+    std::vector<double> tempLoss(size, 0);
+   
+    double power_ratio = 1.0;
+    if (m_tempEnabled) {
+        //use average of the DC voltages to pick which temp curve to use- a weighted average might be better but we don't have that information here
+        double avgDCVoltage = 0;
+        double avgDCPower_Watts = 0;
+        for (size_t i = 0; i < powerDC_Watts_one_inv.size(); i++) {
+            //avgDCVoltage += DCStringVoltage[i];
+            power_ratio = 1.0;
+            calculateTempDerate(DCStringVoltage[i], tempC, powerDC_Watts_one_inv[i], power_ratio, tempLoss[i]);
+        }
+    }
+    // Power quantities go in and come out in units of W
+    double powerAC_Watts = 0;
+    if (m_inverterType == SANDIA_INVERTER || m_inverterType == DATASHEET_INVERTER || m_inverterType == COEFFICIENT_GENERATOR)
+        m_sandiaInverter->acpower(powerDC_Watts_one_inv, DCStringVoltage, &powerAC_Watts, &P_par, &P_lr, &efficiencyAC, &powerClipLoss_kW, &powerConsumptionLoss_kW, &powerNightLoss_kW);
+    else if (m_inverterType == PARTLOAD_INVERTER)
+        m_partloadInverter->acpower(powerDC_Watts_one_inv, &powerAC_Watts, &P_lr, &P_par, &efficiencyAC, &powerClipLoss_kW, &powerNightLoss_kW);
+
+    
+
+    // Scale to total system size
+    // Do not need to scale back up by m_numInverters because scaling them down was a separate vector, powerDC_Watts_one_inv
+    powerDC_kW = 0;
+    double tempLoss_avg = 0;
+    for (size_t i = 0; i < powerDC_Watts_one_inv.size(); i++) {
+        powerDC_kW += powerDC_Watts_one_inv[i] * util::watt_to_kilowatt * m_numInverters;
+        tempLoss_avg += tempLoss[i];
+    }
+    tempLoss_avg /= tempLoss.size();
```

Comment:
The only output that is reported per MPPT input is the DC voltage. Thus, temperature loss is reported as an average